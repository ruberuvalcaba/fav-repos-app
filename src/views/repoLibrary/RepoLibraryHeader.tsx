import React, { useState } from 'react'
import { Thead, Tr, Th } from '@chakra-ui/react'
import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons'
import { Repo } from '../../types'
interface RepoLibraryHeaderProps {
  onSort: (sortKey: keyof Repo) => void
  isAscSortOrder: boolean
}
const RepoLibraryHeader = ({ onSort, isAscSortOrder }: RepoLibraryHeaderProps): JSX.Element => {
  const [activeColumn, setActiveColumn] = useState<string>('')
  const SortIcon = isAscSortOrder ? <ChevronDownIcon /> : <ChevronUpIcon />

  const handleSort = (sortKey: keyof Repo) => {
    setActiveColumn(sortKey)
    onSort(sortKey)
  }
  return (
    <Thead>
      <Tr>
        <Th>Name</Th>
        <Th>Language</Th>
        <Th>Url</Th>
        <Th onClick={() => handleSort('createdAt')} cursor="pointer" w="130px">
          Created at {activeColumn === 'createdAt' && SortIcon}
        </Th>
        <Th onClick={() => handleSort('stargazersCount')} cursor="pointer" w="101px">
          Stars {activeColumn === 'stargazersCount' && SortIcon}
        </Th>
        <Th px={0}></Th>
      </Tr>
    </Thead>
  )
}

export default RepoLibraryHeader
