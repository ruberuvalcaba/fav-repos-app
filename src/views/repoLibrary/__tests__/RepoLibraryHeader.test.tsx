import React from 'react'
import { Table } from '@chakra-ui/react'
import renderer from 'react-test-renderer'
import RepoLibraryHeader from '../RepoLibraryHeader'

it('RepoLibraryHeader component renders correctly', () => {
  const component = renderer.create(
    <Table>
      <RepoLibraryHeader onSort={() => {}} isAscSortOrder={true} />
    </Table>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
