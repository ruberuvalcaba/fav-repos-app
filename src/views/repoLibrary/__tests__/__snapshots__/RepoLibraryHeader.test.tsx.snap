// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoLibraryHeader component renders correctly 1`] = `
<table
  className="chakra-table css-0"
>
  <thead
    className="css-0"
  >
    <tr
      className="css-0"
    >
      <th
        className="css-0"
      >
        Name
      </th>
      <th
        className="css-0"
      >
        Language
      </th>
      <th
        className="css-0"
      >
        Url
      </th>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1wxd1cy{cursor:pointer;width:130px;}",
          }
        }
        data-emotion="css 1wxd1cy"
      />
      <th
        className="css-1wxd1cy"
        onClick={[Function]}
      >
        Created at 
      </th>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-tvbea5{cursor:pointer;width:101px;}",
          }
        }
        data-emotion="css tvbea5"
      />
      <th
        className="css-tvbea5"
        onClick={[Function]}
      >
        Stars 
      </th>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1pom12u{-webkit-padding-start:0px;padding-inline-start:0px;-webkit-padding-end:0px;padding-inline-end:0px;}",
          }
        }
        data-emotion="css 1pom12u"
      />
      <th
        className="css-1pom12u"
      />
    </tr>
  </thead>
</table>
`;
